### PR TITLE
Pkg: upgrade to ginkgo v2, fix #341

### LINKED ISSUE #341: upgrade ginkgo to ginkgo v2
upgrade test case to ginkgo v2

### PR DESCRIPTION
**Pull request by SamYuan1990**
fix #341

Signed-off-by: Sam Yuan <yy19902439@126.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,20 +6,19 @@ require (
...
-	golang.org/x/sys v0.0.0-20220829200755-d48e67d00261
+	golang.org/x/sys v0.1.0
 	k8s.io/api v0.25.0
 	k8s.io/klog/v2 v2.70.1
 )
 
 require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
-	github.com/fsnotify/fsnotify v1.4.9 // indirect
```
Comment (rootfs) [inline go.mod:22]: do you know why some pkgs are removed?

**Comment by rootfs** [comment]
@SamYuan1990 this is a good start! Getting a modern stack helps productivity. 

Can you double check the vendor pkg?

File: go.mod (hunk 1 above)
Comment (SamYuan1990) [reply go.mod:22]:
```
yuanyi@yuanyideMacBook-Pro kepler % go mod why github.com/fsnotify/fsnotify
# github.com/fsnotify/fsnotify
github.com/sustainable-computing-io/kepler/e2e
github.com/sustainable-computing-io/kepler/e2e.test
github.com/onsi/ginkgo
github.com/onsi/ginkgo/internal/remote
github.com/nxadm/tail
github.com/nxadm/tail/watch
github.com/fsnotify/fsnotify
```
above is from ginkgo v1.

```
Switched to branch 'ginkgov2'
yuanyi@yuanyideMacBook-Pro kepler % go mod why github.com/fsnotify/fsnotify
# github.com/fsnotify/fsnotify
(main module does not need package github.com/fsnotify/fsnotify)
```

I suppose in v2 ginkgo removed this package.

Comment (SamYuan1990) [reply go.mod:22]: and I found this commit from ginkgo https://github.com/onsi/ginkgo/commit/102778a4baeb2747386a5b6afa73ebcba5eefe81

**Comment by SamYuan1990** [comment]
> @SamYuan1990 this is a good start! Getting a modern stack helps productivity.
> 
> Can you double check the vendor pkg?

do we have any good idea to check vendor pkg?
it's not a good idea as check file under vendor folder one by one by hash and comparing with original code repo...

personally I wanna to remove vendor folder as I mentioned at https://github.com/sustainable-computing-io/kepler/issues/274

**Comment by SamYuan1990** [comment]
> @SamYuan1990 this is a good start! Getting a modern stack helps productivity.
> 
> Can you double check the vendor pkg?

to make our package management in a modern way, I hope we can considering a case that we need to upgrade any go package in the feature as a general case as our dependence version need bump up...

hence for each time we bump up version of a pkg, we need check go.mod and vendor folder together?

**Comment by rootfs** [comment]
> > @SamYuan1990 this is a good start! Getting a modern stack helps productivity.
> > Can you double check the vendor pkg?
> 
> do we have any good idea to check vendor pkg? it's not a good idea as check file under vendor folder one by one by hash and comparing with original code repo...
> 
> personally I wanna to remove vendor folder as I mentioned at #274

`go mod tidy`

**Comment by SamYuan1990** [comment]
> > > @SamYuan1990 this is a good start! Getting a modern stack helps productivity.
> > > Can you double check the vendor pkg?
> > 
> > 
> > do we have any good idea to check vendor pkg? it's not a good idea as check file under vendor folder one by one by hash and comparing with original code repo...
> > personally I wanna to remove vendor folder as I mentioned at #274
> 
> `go mod tidy`

```
yuanyi@yuanyideMacBook-Pro kepler % git status
On branch ginkgov2
nothing to commit, working tree clean
yuanyi@yuanyideMacBook-Pro kepler % go mod tidy
yuanyi@yuanyideMacBook-Pro kepler % git status
On branch ginkgov2
nothing to commit, working tree clean
```
here you are

**Comment by rootfs** [comment]
sounds good! let's have it

**Review by rootfs — APPROVED** [review]

**Pull request merged**